r2/service.go

```diff
@@ -173,7 +233,13 @@ func NewService(
 	authService auth.HTTPAuthService,
 	store Store,
 ) server.Service {
-	return &service{iOpts: iOpts, store: store, authService: authService, rootPrefix: rootPrefix}
+	return &service{
+		iOpts:          iOpts,
+		store:          store,
+		authService:    authService,
+		rootPrefix:     rootPrefix,
+		serviceMetrics: serviceMetrics{scope: iOpts.MetricsScope()},
```

Comment:
nit: take it or leave it, but the convention is to have a ctor that returns a serviceMetrics struct.